actions/go-flaky-tests/README.md

```diff
@@ -0,0 +1,121 @@
+# Go Flaky Tests
+
+A GitHub Action that detects and analyzes flaky Go tests by fetching logs from Loki.
+
+## Features
+
+- **Loki Integration**: Fetches test failure logs from Loki using LogQL queries
+- **Flaky Test Detection**: Identifies tests that fail inconsistently across different branches
+
+## Usage
+
+```yaml
+name: Go Flaky Tests
+on:
+  schedule:
+    - cron: "0 9 * * 1" # Run every Monday at 9 AM
+  workflow_dispatch:
+
+jobs:
+  analyze-failures:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Go Flaky Tests
+        uses: grafana/shared-workflows/actions/go-flaky-tests@main
+        with:
+          loki-url: ${{ secrets.LOKI_URL }}
+          loki-username: ${{ secrets.LOKI_USERNAME }}
+          loki-password: ${{ secrets.LOKI_PASSWORD }}
+          repository: ${{ github.repository }}
+          time-range: "7d"
+          top-k: "5"
+```
+
+## Inputs
+
+| Input           | Description                                                   | Required | Default |
+| --------------- | ------------------------------------------------------------- | -------- | ------- |
+| `loki-url`      | Loki endpoint URL                                             | ✅       | -       |
+| `loki-username` | Username for Loki authentication                              | ❌       | -       |
+| `loki-password` | Password for Loki authentication                              | ❌       | -       |
```

Comment:
I guess Loki password is a gcom api token right?
Should we also include what permissions should this token have?

Reply:
good call. updated in https://github.com/grafana/shared-workflows/pull/1013/commits/2670ec4390c5a21140fbe9758c7a3e656c868b1d